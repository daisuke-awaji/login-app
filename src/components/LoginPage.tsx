import React, { useState } from 'react'
import { Button, TextField, InputAdornment } from '@material-ui/core'
import { useForm, Controller } from 'react-hook-form'
import { Visibility, VisibilityOff, MailOutline } from '@material-ui/icons'
import { toClicable } from './toClicable'
import { useHistory } from 'react-router-dom'
import { useDispatch } from 'react-redux'
import { AUTHENTICATE } from 'reducers/authenticate'

type Inputs = {
  email: string
  password: string
}

function LoginPage() {
  const { handleSubmit, control, errors, setError } = useForm<Inputs>({
    mode: 'onBlur',
    reValidateMode: 'onChange',
  })

  const history = useHistory()
  const { from }: any = history.location.state || {
    from: { pathname: '/' },
  }
  const dispatch = useDispatch()
  const login = (data: Inputs) => {
    // TODO: 認証APIを実行する
    const valid = (data: Inputs) => {
      const { email, password } = data
      if (email !== 'example@email.com' || password !== 'a') {
        setError(
          'email',
          'hasAnyError',
          'メールアドレスまたはパスワードが不正です。',
        )
        setError(
          'password',
          'hasAnyError',
          'メールアドレスまたはパスワードが不正です。',
        )
        return false
      }
      return true
    }
    if (valid(data)) {
      dispatch({ type: AUTHENTICATE })
      history.replace(from)
    }
  }

  const [visiblePassword, setPasswordVisible] = useState(false)
  const handleClick = () => setPasswordVisible(!visiblePassword)

  return (
    <form onSubmit={handleSubmit(login)}>
      <h3>Login</h3>
      <Controller
        as={
          <TextField
            label="メールアドレス"
            error={!!errors.email}
            variant="outlined"
            margin="dense"
            helperText={errors.email?.message || ''}
            InputProps={{
              startAdornment: (
                <InputAdornment position="start">
                  <MailOutline />
                </InputAdornment>
              ),
            }}
          />
        }
        name="email"
        control={control}
        defaultValue="example@email.com"
        rules={{ required: '必須です。' }}
      />
      <br />
      <Controller
        as={
          <TextField
            label="パスワード"
            error={!!errors.password}
            variant="outlined"
            margin="dense"
            helperText={errors.password?.message || ''}
            type={visiblePassword ? 'default' : 'password'}
            autoComplete="current-password"
            InputProps={{
              startAdornment: (
                <InputAdornment position="start">
                  {visiblePassword
                    ? toClicable(VisibilityOff, handleClick)
                    : toClicable(Visibility, handleClick)}
                </InputAdornment>
              ),
            }}
          />
        }
        name="password"
        control={control}
        // Reactのフォームコンポーネントは、
        // 割り当てられているStateの値がnullかundefinedになると、uncontrolledになってしまうので注意
        defaultValue=""
        rules={{ required: '必須です。' }}
      />
      <br />
      <Controller
        as={
          <Button color="primary">
            <span>Login</span>
          </Button>
        }
        name="submit"
        control={control}
        defaultValue=""
        onClick={handleSubmit(login)}
      />
    </form>
  )
}

export default LoginPage
