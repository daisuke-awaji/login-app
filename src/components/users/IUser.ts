export interface IUser {
  id: string
  name: string
  email: string
  gender?: Gender
  type: UserType
  description?: string
  img?: string
}

export type Gender = 'mail' | 'femail'
export type UserType = 'admin' | 'default'
