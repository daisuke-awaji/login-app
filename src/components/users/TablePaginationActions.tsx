import React from 'react'
import { Theme, createStyles, makeStyles } from '@material-ui/core/styles'
import IconButton from '@material-ui/core/IconButton'
import FirstPageIcon from '@material-ui/icons/FirstPage'
import KeyboardArrowLeft from '@material-ui/icons/KeyboardArrowLeft'
import KeyboardArrowRight from '@material-ui/icons/KeyboardArrowRight'
import LastPageIcon from '@material-ui/icons/LastPage'

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      flexShrink: 0,
      marginLeft: theme.spacing(2.5),
    },
  }),
)
interface TablePaginationActionsProps {
  count: number
  page: number
  rowsPerPage: number
  onChangePage: (
    event: React.MouseEvent<HTMLButtonElement>,
    newPage: number,
  ) => void
}

/**
 *
 * 表の下部に配置する |< < > >| というボタンコンポーネント
 * @param props
 */
export function TablePaginationActions(props: TablePaginationActionsProps) {
  const classes = useStyles()
  const { count, page, rowsPerPage, onChangePage } = props
  return (
    <div className={classes.root}>
      <IconButton
        onClick={(event) => onChangePage(event, 0)}
        disabled={page === 0}
        aria-label="first page |<"
      >
        <FirstPageIcon />
      </IconButton>
      <IconButton
        onClick={(event) => onChangePage(event, page - 1)}
        disabled={page === 0}
        aria-label="previous page <"
      >
        <KeyboardArrowLeft />
      </IconButton>
      <IconButton
        onClick={(event) => onChangePage(event, page + 1)}
        disabled={page >= Math.ceil(count / rowsPerPage) - 1}
        aria-label="next page >"
      >
        <KeyboardArrowRight />
      </IconButton>
      <IconButton
        onClick={(event) => {
          const lastPage = Math.max(0, Math.ceil(count / rowsPerPage) - 1)
          onChangePage(event, lastPage)
        }}
        disabled={page >= Math.ceil(count / rowsPerPage) - 1}
        aria-label="last page >|"
      >
        <LastPageIcon />
      </IconButton>
    </div>
  )
}
