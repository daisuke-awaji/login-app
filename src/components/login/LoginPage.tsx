import React, { useState, useEffect } from 'react'
import {
  Button,
  TextField,
  InputAdornment,
  Typography,
  Link,
  CssBaseline,
  Avatar,
  makeStyles,
  Container,
  FormControlLabel,
  Grid,
  Box,
  Checkbox,
} from '@material-ui/core'
import { useForm, Controller } from 'react-hook-form'
import { Visibility, VisibilityOff, MailOutline } from '@material-ui/icons'
import { toClickable } from './toClickable'
import { useHistory } from 'react-router-dom'
import { useDispatch } from 'react-redux'
import { thunkedLoginAction } from 'reducers/authenticate'
import { Copyright } from './Copyright'
import { AvatarsAnimation } from 'components/AvatarsAnimation'
import { useSelector } from 'react-redux'
import { AppState } from 'reducers/store'

type Inputs = {
  email: string
  password: string
}

const useStyles = makeStyles((theme) => ({
  paper: {
    marginTop: theme.spacing(8),
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
  },
  avatar: {
    margin: theme.spacing(2),
    width: theme.spacing(9),
    height: theme.spacing(9),
    backgroundColor: theme.palette.primary.main,
  },
  form: {
    width: '100%', // Fix IE 11 issue.
    marginTop: theme.spacing(1),
  },
  submit: {
    margin: theme.spacing(3, 0, 2),
  },
  googleLogin: {
    width: '100%',
    marginTop: theme.spacing(1),
  },
}))

const Loading = () => <>Loading...</>

export function LoginPage() {
  const classes = useStyles()

  const [visiblePassword, setPasswordVisible] = useState(false)
  const handleClick = () => setPasswordVisible(!visiblePassword)

  const { handleSubmit, control, errors, setError } = useForm<Inputs>({
    mode: 'onBlur',
    reValidateMode: 'onChange',
  })

  const history = useHistory()

  const dispatch = useDispatch()
  const login = (data: Inputs) => {
    dispatch(thunkedLoginAction(data))
  }
  const auth = useSelector((state: AppState) => state.authenticatedUser)
  useEffect(() => {
    if (auth.user) {
      const { from }: any = history.location.state || {
        from: { pathname: '/' },
      }
      history.replace(from)
    }
    if (auth.error) {
      setError(
        'email',
        'hasAnyError',
        'メールアドレスまたはパスワードが不正です。',
      )
      setError(
        'password',
        'hasAnyError',
        'メールアドレスまたはパスワードが不正です。',
      )
    }
  })
  if (auth.isFetching) {
    return <Loading />
  }

  return (
    <>
      <Container component="main" maxWidth="xs">
        <CssBaseline />
        <div className={classes.paper}>
          <Avatar className={classes.avatar}>
            <AvatarsAnimation />
          </Avatar>
          <Typography component="h1" variant="h5">
            Sign in
          </Typography>
          <form className={classes.form}>
            <Controller
              as={
                <TextField
                  label="メールアドレス"
                  error={!!errors.email}
                  variant="outlined"
                  margin="normal"
                  fullWidth
                  required
                  helperText={errors.email?.message || ''}
                  autoComplete="email"
                  InputProps={{
                    startAdornment: (
                      <InputAdornment position="start">
                        <MailOutline />
                      </InputAdornment>
                    ),
                  }}
                />
              }
              name="email"
              control={control}
              defaultValue=""
              rules={{ required: '必須です。' }}
            />
            <Controller
              as={
                <TextField
                  label="パスワード"
                  error={!!errors.password}
                  variant="outlined"
                  margin="normal" // or dense
                  fullWidth
                  required
                  helperText={errors.password?.message || ''}
                  type={visiblePassword ? 'default' : 'password'}
                  autoComplete="current-password"
                  InputProps={{
                    startAdornment: (
                      <InputAdornment position="start">
                        {visiblePassword
                          ? toClickable(VisibilityOff, handleClick)
                          : toClickable(Visibility, handleClick)}
                      </InputAdornment>
                    ),
                  }}
                />
              }
              name="password"
              control={control}
              // Reactのフォームコンポーネントは、
              // 割り当てられているStateの値がnullかundefinedになると、uncontrolledになってしまうので注意
              // https://github.com/react-hook-form/react-hook-form-website/issues/133
              defaultValue=""
              rules={{ required: '必須です。' }}
            />
            {/* TODO: BELOW */}
            <FormControlLabel
              control={<Checkbox value="remember" color="primary" />}
              label="Remember me"
            />
            <Controller
              as={
                <Button
                  type="submit"
                  fullWidth
                  variant="contained"
                  color="primary"
                  className={classes.submit}
                >
                  Sign In
                </Button>
              }
              name="submit"
              control={control}
              defaultValue=""
              onClick={handleSubmit(login)}
            />
            <Grid container>
              <Grid item xs>
                <Link href="#" variant="body2">
                  Forgot password?
                </Link>
              </Grid>
              <Grid item>
                <Link href="#" variant="body2">
                  {"Don't have an account? Sign Up"}
                </Link>
              </Grid>
            </Grid>
          </form>
        </div>
        <Box mt={8}>
          <Copyright />
        </Box>
      </Container>
    </>
  )
}
