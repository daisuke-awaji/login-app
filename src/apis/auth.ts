import { IUser } from 'components/users/IUser'
import { users } from 'apis/users'
import { sleep } from './utils'

export async function login({ email, password }: any): Promise<IUser> {
  await sleep(500)
  return new Promise((resolve, reject) => {
    if (email === 'usera@email.com' && password === 'a') {
      resolve(users[0])
    } else if (email === 'userb@email.com' && password === 'b') {
      resolve(users[1])
    } else {
      reject(new Error('email or password is not valid'))
    }
  })
}

export async function currentUser(sessionId: string | null): Promise<IUser> {
  await sleep(500)

  return new Promise((resolve, reject) => {
    if (sessionId === '1') {
      resolve(users[0])
    } else if (sessionId === '2') {
      resolve(users[1])
    } else {
      reject(new Error('unauthorized'))
    }
  })
}
