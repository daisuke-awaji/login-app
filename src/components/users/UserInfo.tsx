import React, { useEffect, useState } from 'react'
import { useParams } from 'react-router-dom'
import { fetchUsers } from 'apis/users'
import { IUser } from './IUser'

export const UserInfo = () => {
  // TODO: 本当はユーザ１件取得のAPIを呼ぶ
  const [users, setUsers] = useState<IUser[]>([])
  const fetchData = async () => {
    const users = await fetchUsers()
    setUsers(users)
  }
  useEffect(() => {
    fetchData()
  }, [])

  const { id } = useParams()
  const one = users.find((user) => user.id === Number(id))
  return (
    <div>
      <h1>{one?.name}</h1>
      <h3>Email: {one?.email}</h3>
    </div>
  )
}
