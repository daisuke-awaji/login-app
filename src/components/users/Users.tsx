import React, { useEffect, useState } from 'react'
import { Link } from 'react-router-dom'
import { fetchUsers } from 'apis/users'

import Table from '@material-ui/core/Table'
import TableBody from '@material-ui/core/TableBody'
import TableCell from '@material-ui/core/TableCell'
import TableContainer from '@material-ui/core/TableContainer'
import TableHead from '@material-ui/core/TableHead'
import TableRow from '@material-ui/core/TableRow'
import Paper from '@material-ui/core/Paper'
import Typography from '@material-ui/core/Typography'
import { IUser } from './IUser'

export const Users = () => {
  const [users, setUsers] = useState<IUser[]>([])

  const fetchData = async () => {
    const users = await fetchUsers()
    setUsers(users)
  }
  useEffect(() => {
    fetchData()
  }, [])
  return (
    <>
      <Typography variant="h3">Users</Typography>
      <TableContainer component={Paper}>
        <Table style={{ tableLayout: 'auto' }} aria-label="user table">
          <TableHead>
            <TableRow>
              <TableCell style={{ width: '10%' }} align="right">
                Id
              </TableCell>
              <TableCell style={{ width: '30%' }} align="right">
                Name
              </TableCell>
              <TableCell style={{ width: '40%' }} align="right">
                Email
              </TableCell>
              <TableCell style={{ width: '10%' }} align="right">
                Type
              </TableCell>
            </TableRow>
          </TableHead>
          <TableBody>
            {users.map((user) => (
              <TableRow key={user.name}>
                <TableCell style={{ width: '10%' }} align="right">
                  {user.id}
                </TableCell>
                <TableCell style={{ width: '30%' }} align="right">
                  <Link to={'/users/' + user.id}>{user.name}</Link>
                </TableCell>
                <TableCell
                  style={{
                    width: '40%',
                    maxWidth: '100px',
                    // textOverflow: 'ellipsis', // はみ出たテキストを切り取って ... を表示する
                    whiteSpace: 'nowrap',
                    overflow: 'scroll',
                  }}
                  align="right"
                >
                  {user.email}
                </TableCell>
                <TableCell style={{ width: '10%' }} align="right">
                  {user.type}
                </TableCell>
              </TableRow>
            ))}
          </TableBody>
        </Table>
      </TableContainer>
    </>
  )
}
