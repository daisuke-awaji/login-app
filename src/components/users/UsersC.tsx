import React from 'react'
import { Link } from 'react-router-dom'
import { fetchUsers } from 'apis/users'
import { IUser } from './IUser'

import Table from '@material-ui/core/Table'
import TableBody from '@material-ui/core/TableBody'
import TableCell from '@material-ui/core/TableCell'
import TableContainer from '@material-ui/core/TableContainer'
import TableHead from '@material-ui/core/TableHead'
import TableRow from '@material-ui/core/TableRow'
import Paper from '@material-ui/core/Paper'
import Typography from '@material-ui/core/Typography'

interface IProps {
  classes?: any
}
interface IState {
  users: IUser[]
}

export class Users extends React.Component<IProps, IState> {
  constructor(props: any) {
    super(props)
    this.state = { users: [] }
  }
  async componentDidMount() {
    const users = await fetchUsers()
    this.setState({ users: users })
  }
  render() {
    return (
      <>
        <Typography variant="h3">Users</Typography>
        <TableContainer component={Paper}>
          <Table style={{ tableLayout: 'auto' }} aria-label="user table">
            <TableHead>
              <TableRow>
                <TableCell style={{ width: '10%' }} align="right">
                  Id
                </TableCell>
                <TableCell style={{ width: '30%' }} align="right">
                  Name
                </TableCell>
                <TableCell style={{ width: '40%' }} align="right">
                  Email
                </TableCell>
                <TableCell style={{ width: '10%' }} align="right">
                  Type
                </TableCell>
              </TableRow>
            </TableHead>
            <TableBody>
              {this.state.users.map((user) => (
                <TableRow key={user.name}>
                  <TableCell style={{ width: '10%' }} align="right">
                    {user.id}
                  </TableCell>
                  <TableCell style={{ width: '30%' }} align="right">
                    <Link to={'/users/' + user.id}>{user.name}</Link>
                  </TableCell>
                  <TableCell
                    style={{
                      width: '40%',
                      maxWidth: '100px',
                      // textOverflow: 'ellipsis', // はみ出たテキストを切り取って ... を表示する
                      whiteSpace: 'nowrap',
                      overflow: 'scroll',
                    }}
                    align="right"
                  >
                    {user.email}
                  </TableCell>
                  <TableCell style={{ width: '10%' }} align="right">
                    {user.type}
                  </TableCell>
                </TableRow>
              ))}
            </TableBody>
          </Table>
        </TableContainer>
      </>
    )
  }
}
