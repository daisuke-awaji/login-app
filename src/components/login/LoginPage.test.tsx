import React from 'react'
import { Provider } from 'react-redux'
import { Router } from 'react-router-dom'
import { combineReducers, createStore } from 'redux'
import auth, { loginAction } from 'reducers/authenticate'
import { LoginPage } from './LoginPage'
import renderer from 'react-test-renderer'
import { IUser } from '../users/IUser'

const setup = () => {
  // Routerコンポーネントに依存する hisotry は mock 化して注入する
  const mockHistory: any = { push: jest.fn(), location: {}, listen: jest.fn() }
  // Redux は必要な store だけ mock を作成して注入する
  const reducer = combineReducers({ authenticatedUser: auth })
  const mockStore = createStore(reducer)
  const component = (
    <Provider store={mockStore}>
      <Router history={mockHistory}>
        <LoginPage />
      </Router>
    </Provider>
  )

  const user: IUser = {
    id: '1',
    name: 'John Scott',
    email: 'john@email.com',
    type: 'admin',
  }
  mockStore.dispatch(loginAction(user))
  return { component }
}

describe('スナップショット', () => {
  it('レンダリングができること', () => {
    const { component } = setup()
    const header = renderer.create(component)
    let tree = header.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
