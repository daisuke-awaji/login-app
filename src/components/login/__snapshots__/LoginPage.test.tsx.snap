// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`スナップショット レンダリングができること 1`] = `
<main
  className="MuiContainer-root MuiContainer-maxWidthXs"
>
  <div
    className="makeStyles-paper-1"
  >
    <div
      className="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-2 MuiAvatar-colorDefault"
    >
      <div
        aria-label="animation"
        onClick={[Function]}
        role="button"
        style={
          Object {
            "height": "100%",
            "margin": "0 auto",
            "outline": "none",
            "overflow": "hidden",
            "width": "100%",
          }
        }
        tabIndex="0"
        title=""
      />
    </div>
    <h1
      className="MuiTypography-root MuiTypography-h5"
    >
      Sign in
    </h1>
    <form
      className="makeStyles-form-3"
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-required Mui-required"
          data-shrink={true}
        >
          メールアドレス
          <span
            aria-hidden={true}
            className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart MuiOutlinedInput-adornedStart"
          onClick={[Function]}
        >
          <div
            className="MuiInputAdornment-root MuiInputAdornment-positionStart"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 14H4V8l8 5 8-5v10zm-8-7L4 6h16l-8 5z"
              />
            </svg>
          </div>
          <input
            aria-invalid={false}
            autoComplete="email"
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart"
            disabled={false}
            name="email"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="text"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-125 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-127 PrivateNotchedOutline-legendNotched-128"
            >
              <span>
                メールアドレス
                 *
              </span>
            </legend>
          </fieldset>
        </div>
        
      </div>
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-required Mui-required"
          data-shrink={true}
        >
          パスワード
          <span
            aria-hidden={true}
            className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart MuiOutlinedInput-adornedStart"
          onClick={[Function]}
        >
          <div
            className="MuiInputAdornment-root MuiInputAdornment-positionStart"
          >
            <span
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                />
              </svg>
            </span>
          </div>
          <input
            aria-invalid={false}
            autoComplete="current-password"
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart"
            disabled={false}
            name="password"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="password"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-125 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-127 PrivateNotchedOutline-legendNotched-128"
            >
              <span>
                パスワード
                 *
              </span>
            </legend>
          </fieldset>
        </div>
        
      </div>
      <label
        className="MuiFormControlLabel-root"
      >
        <span
          aria-disabled={false}
          className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-141 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <span
            className="MuiIconButton-label"
          >
            <input
              className="PrivateSwitchBase-input-144"
              data-indeterminate={false}
              onChange={[Function]}
              type="checkbox"
              value="remember"
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
          </span>
        </span>
        <span
          className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
        >
          Remember me
        </span>
      </label>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-4 MuiButton-containedPrimary MuiButton-fullWidth"
        disabled={false}
        name="submit"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
        value=""
      >
        <span
          className="MuiButton-label"
        >
          Sign In
        </span>
      </button>
      <div
        className="MuiGrid-root MuiGrid-container"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
        >
          <a
            className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary"
            href="#"
            onBlur={[Function]}
            onFocus={[Function]}
          >
            Forgot password?
          </a>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <a
            className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary"
            href="#"
            onBlur={[Function]}
            onFocus={[Function]}
          >
            Don't have an account? Sign Up
          </a>
        </div>
      </div>
    </form>
  </div>
  <div
    className="MuiBox-root MuiBox-root-296"
  >
    <p
      className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
    >
      Copyright © 
      <a
        className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
        href="https://material-ui.com/"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        Login app
      </a>
       
      2020
      .
    </p>
  </div>
</main>
`;
